lib/saluki-env/src/autodiscovery/mod.rs

```diff
@@ -236,4 +296,56 @@ mod tests {
         assert_eq!(svc.name, "nginx");
         assert_eq!(svc.namespace, "default");
     }
+
+    #[test]
+    fn test_autodiscovery_event_from_proto_config() {
+        // Create a ProtoConfig with test values
+        let mut proto_config = ProtoConfig {
+            name: "test-config".to_string(),
+            event_type: ConfigEventType::Schedule as i32,
+            init_config: b"init-data".to_vec(),
+            instances: vec![b"instance1".to_vec(), b"instance2".to_vec()],
+            provider: "test-provider".to_string(),
+            ad_identifiers: vec!["id1".to_string(), "id2".to_string()],
+            cluster_check: true,
+            metric_config: vec![],
+            logs_config: vec![],
+            advanced_ad_identifiers: vec![],
+            service_id: "service-id".to_string(),
+            tagger_entity: "tagger-entity".to_string(),
+            node_name: "node-name".to_string(),
+            source: "source".to_string(),
+            ignore_autodiscovery_tags: false,
+            metrics_excluded: false,
+            logs_excluded: false,
+        };
+
+        let kube_svc = KubeNamespacedName {
+            name: "nginx".to_string(),
+            namespace: "default".to_string(),
+        };
+
+        let adv_id = AdvancedAdIdentifier {
+            kube_service: Some(kube_svc),
+            kube_endpoints: None,
+        };
+
+        proto_config.advanced_ad_identifiers = vec![adv_id];
+
+        let event = AutodiscoveryEvent::from(proto_config.clone());
+
+        match event {
+            AutodiscoveryEvent::Schedule { config: _config } => {}
+            _ => panic!("Expected an Schedule event"),
+        }
```

Comment:
There's a handy macro, `matches!`, which returns a boolean value based on if the pattern match is a hit or not:

```suggestion
        assert!(matches!(event, AutodiscoveryEvent::Schedule { .. }), "expected schedule event");
```

We're also taking advantage of the ability to ignore unbound fields of the enum variant by using `..`. This is the equivalent of doing `foo: _` for every field in the variant.

Reply:
Thanks for the tip